File: app/src/main/java/com/v2/desafionubank/di/module/NavigationModule.java
package com.v2.desafionubank.di.module;

import com.v2.desafionubank.ui.NavigationManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by CaioSChristino on 20/02/2018.
 */

@Module
public class NavigationModule {

    @Provides
    @Singleton
    protected NavigationManager provideNavigationManager() {
        return new NavigationManager();
    }
}

File: app/src/main/java/com/v2/desafionubank/di/module/NavigationModule.kt
package com.v2.desafionubank.di.module

import com.v2.desafionubank.ui.NavigationManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by csanchez on 19/04/2018.
 */

@Module
class NavigationModule {
    @Provides
    @Singleton
    protected fun provideNavigationManager(): NavigationManager {
        return NavigationManager()
    }
}